src/test/java/org/jenkinsci/plugins/docker/workflow/WithContainerStepTest.java

```diff
@@ -185,8 +185,8 @@ public class WithContainerStepTest {
                 WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "prj");
                 p.setDefinition(new CpsFlowDefinition(
                     "node {\n" +
-                    "  withDockerContainer('httpd:2.4.12') {\n" +
-                    "    sh \"sleep 5; ps -e -o pid,command | egrep '${pwd tmp: true}/durable-.+/script.sh' | fgrep -v grep | sort -n | tr -s ' ' | cut -d ' ' -f2 | xargs kill -9\"\n" +
+                    "  withDockerContainer('httpd:2.4.54-alpine') {\n" +
+                    "    sh \"set -e; sleep 5; ps -e -o pid,args | egrep '${pwd tmp: true}/durable-.+/script.sh' | fgrep -v grep | sort -n | tr -s ' ' | cut -d ' ' -f2 | xargs kill -9\"\n" +
```

Comment:
doing this whole parsing looks so painful to me 😢 
Would it be easier to test if the heartbeat file still existed/was being written to?

Reply:
I think this test is trying to check whether `sh` steps inside of `withContainer` respond correctly if the remote script just suddenly dies, so it is using internal knowledge of `durable-task` to kill all of the processes related to the `sh` step from within, which is pretty awkward. It used to be simpler before https://github.com/jenkinsci/durable-task-plugin/pull/49, but it had to be updated in https://github.com/jenkinsci/docker-workflow-plugin/pull/121 since the `pid` file didn't exist anymore.

I think that if we deleted the heartbeat file here it wouldn't do anything, since the control script would just touch the file 3 seconds later, recreating it. Not 100% sure though. Maybe we could `chmod` it to be unwritable or something, but IDK if that would be much easier to maintain.